### PR TITLE
Global support for datacite 4.0

### PR DESCRIPTION
**Pull request by remileduc**
FIX https://github.com/inveniosoftware/datacite/issues/15

_done_:
- create `schema40.py`
- create JSON schema for v4.0
- create basic tests for v4.0
- change the tests for specific fields in v4.0
- update documentation

_not done yet_:
- refactor code to factorize common rules between v3.1 and v4.0
  -> should it be part of another PR?

_what is done_:
Fully working and tested support for v4.0 (main test with full JSON example works).

_FIX_
- affiliation field is now an array for both creators and contributors

_Hint for CR_
On this PR, the changelog of github is not very nice... You can manually look at the diffs between the following files to see what has been modified:
- `schema31.py` and `schema40.py`
- `datacite-v3.1.json` and `datacite-v4.0.json`
- `datacite-v3.1-full-example.json` and `datacite-v4.0-full-example.json`
- `test_schema31` and `test_schema40`

### CODE REVIEW

**Comment by jirikuncar** [comment]
@remileduc the refactoring can be done later if we have good test coverage.

**Comment by lnielsen** [comment]
@remileduc I agree with @jirikuncar. Also, "refactor tests to factorize common rules" is not needed. I think it's actually better to have tests for 4.0 and 3.1 split.

File: tests/example/full.py [hunk 1]
```diff
@@ -13,13 +16,16 @@
     ],
     'publisher': 'CERN',
     'publicationYear': '2015',
+    'resourceType': {
+        'resourceTypeGeneral': 'Dataset'
+    }
 }
 
 # Validate dictionary
-assert schema31.validate(data)
```
Comment (remileduc) [inline tests/example/full.py]: In any case, this test is still run here: https://github.com/inveniosoftware/datacite/blob/master/tests/test_schema31.py#L34
But I can create another file for the example. However, except for the `resourceType` that became mandatory in v4.0, there will be no differences between the 2 examples :/

Comment (jirikuncar) [inline tests/example/full.py]: Thanks for explanation. I am fine to keep it as is in this case. However we should be always careful with modification of existing test cases.

**Comment by lnielsen** [comment]
LGTM

**Pull request merged**